### PR TITLE
Annotate System.ServiceProcess.ServiceController for nullable reference types

### PR DESCRIPTION
**Pull request by stephentoub**
Contributes to https://github.com/dotnet/runtime/issues/2339
cc: @Anipik, @buyaa-n

### CODE REVIEW

File: src/libraries/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs [hunk 1]
```diff
@@ -25,17 +26,17 @@ public class ServiceBase : Component
...
-        private ServiceMainCallback _mainCallback;
-        private ManualResetEvent _startCompletedSignal;
-        private ExceptionDispatchInfo _startFailedException;
+        private ServiceControlCallbackEx? _commandCallbackEx;
+        private ServiceMainCallback? _mainCallback;
+        private ManualResetEvent? _startCompletedSignal;
+        private ExceptionDispatchInfo? _startFailedException;
         private int _acceptedCommands;
-        private string _serviceName;
+        private string? _serviceName;
```
Comment (buyaa-n) [inline src/libraries/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs:34]: This does not seem to be nullable, set in constructor with "" using ServiceName property, and ServiceName property not allowing null for set

File: src/libraries/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs [hunk 2]
```diff
@@ -263,7 +264,8 @@ protected IntPtr ServiceHandle
         /// <devdoc>
         ///    <para> Indicates the short name used to identify the service to the system.</para>
         /// </devdoc>
-        public string ServiceName
+        [DisallowNull]
+        public string? ServiceName
```
Comment (buyaa-n) [inline src/libraries/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs:268]: With above I think it should be non null

**Review by buyaa-n — APPROVED** [review]
One question for `ServiceName` other than that looks good, thanks!

File: src/libraries/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs]: Good catch. Fixed.

**Pull request merged**